ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/BlockAddedEvent.java

```diff
@@ -55,16 +58,23 @@ public static BlockAddedEvent createForHeadAdvancement(
         block,
         block.getBody().getTransactions(),
         Collections.emptyList(),
-        logsWithMetadata);
+        logsWithMetadata,
+        null);
   }
 
   public static BlockAddedEvent createForChainReorg(
       final Block block,
       final List<Transaction> addedTransactions,
       final List<Transaction> removedTransactions,
-      final List<LogWithMetadata> logsWithMetadata) {
+      final List<LogWithMetadata> logsWithMetadata,
+      final Block reorgBlock) {
```

Comment:
`reorgBlock` is a bit ambiguous with `block` above. I suggest renaming the first one to `newChainHead` and the second to `commonAncestorWithOldHead`

Reply:
I think there's actually always a sensible value for `commonAncestorWithOldHead`, even for non-reorgs. It would be the parent block of the block in the event. Then, the new block subscriptions could actually just walk between the common ancestor block and the event block, sending notifcations as they go. I think this would remove the need to split the logic in `onBlockAdded`.